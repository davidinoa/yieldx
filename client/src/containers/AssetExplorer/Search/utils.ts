interface InGridFilter {
  filter: string | number;
  filterTo: number;
  filterType: string;
  type: string;
  dateFrom: string;
  dateTo: string;
  values: string[];
}

export interface AgGridFilters {
  [key: string]: any;
}

const getFilter = (filter: InGridFilter) => {
  /**
   *  The filter object from Ag Grid API. In the case of filtering `name` column.
   *  { filter: "Asset Name", filterType: "text", type: "contains"}
   */
  let searchFilter = {};
  const {
    filter: filterContent,
    filterType,
    type,
    filterTo,
    values,
    dateFrom,
    dateTo
  } = filter;
  if (filterType === "text") {
    switch (type) {
      case "contains":
        searchFilter = { containsText: filterContent };
        break;
      case "notContains":
        searchFilter = { doesNotContainText: filterContent };
        break;
      case "equals":
        searchFilter = { equalsText: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualText: filterContent };
        break;
      case "startsWith":
        searchFilter = { startsWithText: filterContent };
        break;
      case "endsWith":
        searchFilter = { endsWith: filterContent };
        break;
      default:
        searchFilter = { containsText: filterContent };
        break;
    }
  } else if (filterType === "number") {
    switch (type) {
      case "equals":
        searchFilter = { equalsNumber: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualNumber: filterContent };
        break;
      case "lessThan":
        searchFilter = { isLessThanNumber: filterContent };
        break;
      case "lessThanOrEqual":
        searchFilter = { isLessThanOrEqualToNumber: filterContent };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanNumber: filterContent };
        break;
      case "greaterThanOrEqual":
        searchFilter = { isGreaterThanOrEqualToNumber: filterContent };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToNumber: filterContent,
          isLessThanOrEqualToNumber: filterTo
        };
        break;
      default:
        searchFilter = { equalsNumber: filterContent };
        break;
    }
  } else if (filterType === "date") {
    const convertedDateFrom = new Date(dateFrom);
    const convertedDateTo = new Date(dateTo);
    switch (type) {
      case "equals":
        searchFilter = { equalsDate: convertedDateFrom };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanDate: convertedDateFrom };
        break;
      case "lessThan":
        searchFilter = { isLessThanDate: convertedDateFrom };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualDate: convertedDateFrom };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToDate: convertedDateFrom,
          isLessThanOrEqualToDate: convertedDateTo
        };
        break;
      default:
        searchFilter = { equalsDate: convertedDateFrom };
        break;
    }
  } else if (filterType === "set" && values) {
    // We may need to add `set` type Ag grid column filter params like cusip or isin
    // in the search request in the future.
    // searchFilter = { inText: values };
  }
  return searchFilter;
};

export const convertInGridFilters = (filters: AgGridFilters) => {
  const convertedFilters: { [index: string]: any } = {};
  Object.keys(filters).forEach(key => {
    const inGridFilter = filters[key];
    if (key === "minimumQuantity") key = "bond.minimumDenomination";
    convertedFilters[key] = getFilter(inGridFilter);
  });

  return convertedFilters;
};
