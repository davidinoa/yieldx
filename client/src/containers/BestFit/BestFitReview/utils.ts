import { getRatingValue } from "../../../util/ratings";
import YieldXPortfolio, {
  YieldXPosition
} from "../../../models/YieldXPortfolio";

export function getBondRatingsData(
  portfolio: YieldXPortfolio
): { name: string; value: number }[] {
  if (portfolio.getBondSleeve) {
    const ratingMarketValues: { [key: string]: number } = {};
    const bondPortfolio = portfolio.getBondSleeve();
    bondPortfolio.positions.forEach((position: YieldXPosition) => {
      const { asset, weight } = position;
      const rating = asset.rating ?? "NR";
      ratingMarketValues[rating] = (ratingMarketValues[rating] || 0) + weight;
    });
    return Object.entries(ratingMarketValues)
      .map(([rating, weight]) => {
        return {
          name: rating,
          value: weight
        };
      })
      .sort((a, b) => {
        return getRatingValue(a.name) - getRatingValue(b.name);
      });
  }
  return [];
}
