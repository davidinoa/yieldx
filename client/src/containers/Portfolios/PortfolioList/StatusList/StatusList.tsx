import React from "react";
import {
  ListStatuses,
  Status
} from "../../../../components/ListPages/ListPages.styles";

type Props = {
  statusCount: {
    active: number;
    closed: number;
    tracking: number;
  };
};

export default function StatusList({ statusCount }: Props) {
  return (
    <ListStatuses>
      <span>
        <Status className="ready" />
        <b>active:</b>
        {statusCount.active}
      </span>
      <span>
        <Status className="terminated" />
        <b>closed:</b>
        {statusCount.closed}
      </span>
      <span>
        <Status className="tracking" />
        <b>tracking:</b>
        {statusCount.tracking}
      </span>
    </ListStatuses>
  );
}
