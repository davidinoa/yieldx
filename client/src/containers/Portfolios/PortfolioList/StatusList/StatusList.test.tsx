import React from "react";
import { render } from "test/utils";
import StatusList from "./StatusList";

describe("StatusList", () => {
  it("Should match snapshot", () => {
    const { asFragment } = render(
      <StatusList statusCount={{ active: 0, closed: 0, tracking: 100 }} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
