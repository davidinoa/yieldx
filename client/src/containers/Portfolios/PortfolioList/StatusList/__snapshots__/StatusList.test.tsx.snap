// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusList Should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 0.35rem;
  padding-bottom: 0.35rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
  position: relative;
  padding-right: 0.75rem;
  margin-right: 0.75rem;
  font-size: 0.65rem;
}

.c0 span i {
  display: inline-block;
  margin-right: 0.25rem;
}

.c0 span b {
  margin-right: 5px;
  margin-left: 5px;
  font-weight: bold;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  text-transform: uppercase;
  color: var(--subtext-3);
}

.c0 span::after {
  content: "";
  border-right: solid 1px var(--light);
  width: 2px;
  height: 100%;
  position: absolute;
  top: 0;
  right: 0;
}

.c0 span:last-of-type::after {
  border: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  width: 5px;
  border-radius: 50%;
  margin: 0 auto;
}

.c1.active,
.c1.ready {
  background: var(--active);
}

.c1.deactivated,
.c1.inactive,
.c1.blocked,
.c1.closed,
.c1.terminated {
  background: var(--subtext-3);
}

.c1.tracking {
  background: var(--tracking);
}

.c1.pending {
  background: var(--caution);
}

.c1.error {
  background: var(--issue);
}

@media (min-width:1600px) {
  .c0 {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }
}

@media (min-width:1600px) {
  .c0 span {
    padding-right: 1.25rem;
    margin-right: 1.25rem;
    font-size: 0.75rem;
  }
}

@media (min-width:1600px) {
  .c1 {
    height: 6px;
    width: 6px;
  }
}

<div
    class="c0 col-17 col-xxl-16"
  >
    <span>
      <i
        class="c1 ready"
      />
      <b>
        active:
      </b>
      0
    </span>
    <span>
      <i
        class="c1 terminated"
      />
      <b>
        closed:
      </b>
      0
    </span>
    <span>
      <i
        class="c1 tracking"
      />
      <b>
        tracking:
      </b>
      100
    </span>
  </div>
</DocumentFragment>
`;
